hazelcast/src/main/java/com/hazelcast/internal/plugin/multicast/MulticastDiscoveryReceiver.java

```
@@ -0,0 +1,69 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.plugin.multicast;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.net.DatagramPacket;
+import java.net.MulticastSocket;
+
+public class MulticastDiscoveryReceiver {
+
+    MulticastSocket multicastSocket;
+    private static final int DATAGRAM_BUFFER_SIZE = 64 * 1024;
+    DatagramPacket datagramPacketReceive = new DatagramPacket(new byte[DATAGRAM_BUFFER_SIZE], DATAGRAM_BUFFER_SIZE);
+
+
+    public MulticastDiscoveryReceiver(MulticastSocket multicastSocket) {
+        this.multicastSocket = multicastSocket;
+    }
+
+    public MemberInfo receive() {
+        try {
+            Object o;
+            multicastSocket.receive(datagramPacketReceive);
+            byte[] data = datagramPacketReceive.getData();
+            MemberInfo memberInfo = null;
+            ByteArrayInputStream bis = new ByteArrayInputStream(data);
+            ObjectInput in = null;
+            try {
+                in = new ObjectInputStream(bis);
+                o = in.readObject();
+                memberInfo = (MemberInfo) o;
+            } finally {
+                try {
+                    bis.close();
+                } catch (IOException ex) {
+                    // ignore close exception
+                }
+                try {
+                    if (in != null) {
+                        in.close();
+                    }
+                } catch (IOException ex) {
+                    // ignore close exception
+                }
+                return memberInfo;
+            }
+
+        } catch (Exception e) {
```

Here all exceptions are being swallowed; could make debugging a problem very hard.